server/service/src/processors/transfer/shipment/create_inbound_shipment.rs

```diff
@@ -152,6 +152,13 @@ fn generate_inbound_shipment(
         None => format!("Stock transfer"),
     };
 
+    // If outbound shipment has tax, then we will also have tax in inbound shipment
+    // but greater than 0.0
+    let tax = match outbound_shipment_row.tax {
+        Some(tax) if tax > 0.0 => Some(tax),
+        _ => None,
+    };
+
```

Comment:
The actual issue was when the outbound shipment had no tax value (`None`) to be calculated for its linked inbound shipment, `outbound_shipment.tax` syncs to the central server with `null` value but the `transact.tax` in the central server is of `Real` typed so when saving record `null` converted to `0` automatically and then sync as `0` to the receiving site. But when inbound shipment is generated, an extra invoice line tries to get created cause of,
https://github.com/msupply-foundation/open-msupply/blob/1a203521ba589ef633386e2a52330a07374e2578/server/service/src/invoice/inbound_shipment/update/mod.rs#L92-L96
which is related to,
https://github.com/msupply-foundation/open-msupply/blob/1a203521ba589ef633386e2a52330a07374e2578/server/service/src/invoice/inbound_shipment/update/generate.rs#L93-L103

I presume if tax is configured (i.e. in the central server atm) then it should be > 0 or if it 0 then would be considered as (not being configured yet)?

Reply:
Tax in omSupply is actually (tax_rate), so actually different from how tax works in mSupply. There is an existing migration issues for this. #3018
This change actually doesn't do anything because if outbound.tax is None then it should be None when transferring to inbound and shouldn't get sent when syncing to central

Reply:
@raviSussol 

> an extra invoice line tries to get created cause of,
> which is related to

To me, the issue is we are trying to update the line twice, and overwriting the first with the second, we had similar issue previously, we need to revert to using invoice_line_repository.update_tax

Reply:
Would be great to have a test to capture this btw

Reply:
> @raviSussol
> 
> > an extra invoice line tries to get created cause of,
> > which is related to
> 
> To me, the issue is we are trying to update the line twice, and overwriting the first with the second, we had similar issue previously, we need to revert to using invoice_line_repository.update_tax

Probably for the best~

Reply:
Thanks @andreievg - have updated changes.